app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,41 @@
+class CommentsController < ApplicationController
+
+  before_action :authenticate_user!
+
+  def new
+    @image = Image.find(params[:image_id])
+    @comment = Comment.new
+  end
+
+  def create
+    @image = Image.find(params[:image_id])
+    @comment = @image.comments.create_with_user(comment_params, current_user.username)
+    redirect_to images_path
+  end
+
+  def comment_params
+    params.require(:comment).permit(:comments)
+  end
+
+  def edit
+    @comment = Comment.find(params[:id])
+    if current_user.username == @comment.username
+      render 'edit'
+    else
+      flash[:notice] = "You can not edit someone else's comment"
+      redirect_to '/images'
+    end
+  end
+
+  def destroy
+    @comment = Comment.find(params[:id])
+    if current_user.username == @comment.username
```

Comment:
Refactor this kind of logic to the model. In this case, comment. Check this out: https://github.com/makersacademy/slack-overflow/issues/15